secure_strings.tf

```diff
@@ -1,23 +1,39 @@
 locals {
-  secure_env_vars_count = length(keys(var.secure_env_vars)) > 0 ? 1 : 0
-  secure_env_vars_exports = local.secure_env_vars_count > 0 ? join(
+  has_secure_env_vars = var.secure_env_vars != null && length(var.secure_env_vars) > 0
+
+  secure_env_vars_exports = join(
     "\n",
     [
       for key, _ in var.secure_env_vars : "export ${key}=$(echo $SECRET_VALUE | jq -r '.${key}')"
     ]
-  ) : ""
+  )
+
   secure_env_vars = "export SECRET_VALUE=$(aws secretsmanager get-secret-value --secret-id ${local.base_name}-secret --query SecretString --output text)\n${local.secure_env_vars_exports}"
 }
 
+resource "validation_warning" "token_or_private_key_in_plaintext" {
+  condition = strcontains(var.configuration, "export SPACELIFT_TOKEN") || strcontains(var.configuration, "export SPACELIFT_POOL_PRIVATE_KEY")
+  summary   = "Detected sensitive environment variable in plaintext format"
+  details   = <<EOT
+The 'configuration' parameter seems to contain the 'SPACELIFT_TOKEN' or 'SPACELIFT_POOL_PRIVATE_KEY' environment variables.
+These configuration values are injected in plaintext format into the user data script.
+It is highly recommended to use the 'secure_env_vars' parameter to store sensitive information.
+EOT
+}
```

Comment:
@Apollorion @adamconnelly here's the warning logic

Reply:
yeah, it's already called out in the README. idk if people look at it much though. 🤔